# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


In [1]:
import pandas  as pd
import numpy as np

In [2]:
car=pd.read_csv('data/car-sales.csv',header=0)

### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [5]:
car.head(10)

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

In [15]:
# sắp xếp price theo thứ tự giảm dần và hiển thị 5 xe có giá cao nhất
car.sort_values(by='Price',ascending=False).head(5)

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0
2       Toyota   Blue     32549      3   7000.0
7        Honda   Blue     54738      4   7000.0

In [22]:
# Tính tỉ lệ phần trăm của xe Toyota theo price

ti=car[car['Manufacturer']=='Toyota']['Price'].mean()/car['Price'].mean()
ti*100

71.1249182472204

NameError: name 'm' is not defined

In [5]:
car.shape

(10, 5)

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [9]:
car.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Manufacturer  10 non-null     object 
 1   Colour        10 non-null     object 
 2   Odometer      10 non-null     int64  
 3   Doors         10 non-null     int64  
 4   Price         10 non-null     float64
dtypes: float64(1), int64(2), object(2)
memory usage: 532.0+ bytes


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [12]:
car['Manufacturer'].count()

10

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [15]:
car['Manufacturer'].value_counts()

Manufacturer
Toyota    4
Honda     3
Nissan    2
BMW       1
Name: count, dtype: int64

### 5. Đếm số xe theo màu


In [16]:
car['Colour'].value_counts()

Colour
White    4
Blue     3
Red      1
Black    1
Green    1
Name: count, dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [17]:
car['Price'].sum()

76450.0

### 7. Top 3 xe có giá cao nhất


In [20]:
car.sort_values(by='Price',ascending=False).head(3)

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [21]:
car.sort_values(by='Odometer',ascending=False).head(3)

Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [24]:
honda=car[car['Manufacturer']=='Honda']
# - Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
print("Tỷ lệ xe Honda trong dữ liệu của cửa hàng là: ",honda.shape[0]/car.shape[0])
# - Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
print("Giá trung bình của xe hãng Honda của cửa hàng là: ",honda['Price'].mean())
# - Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)
print("Các xe của hãng Honda trong cửa hàng có những màu sắc là: ",honda['Colour'].unique())

Tỷ lệ xe Honda trong dữ liệu của cửa hàng là:  0.3
Giá trung bình của xe hãng Honda của cửa hàng là:  6500.0
Các xe của hãng Honda trong cửa hàng có những màu sắc là:  ['Red' 'Blue']


# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [30]:
# - Giá trung bình các xe (7645)
print("Giá trung bình các xe là: ",car['Price'].mean())
# - Giá trị trung vị các xe (6625)
print("Giá trị trung vị các xe là: ",car['Price'].median())
# - Miền giá trị của xe (18500)
print("Miền giá trị của xe là: ",car['Price'].max()-car['Price'].min())
# - Miền phân vị của xe (2750)
print("Miền phân vị của xe là: ",car['Price'].quantile(0.75)-car['Price'].quantile(0.25))
# - Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])
print("Điểm phân vị 0.4 và 0.6 của giá là: ",[car['Price'].quantile(0.4),car['Price'].quantile(0.6)])

Giá trung bình các xe là:  7645.0
Giá trị trung vị các xe là:  6625.0
Miền giá trị của xe là:  18500.0
Miền phân vị của xe là:  2750.0
Điểm phân vị 0.4 và 0.6 của giá là:  [5750.0, 7000.0]


---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

In [31]:
tip=pd.read_csv('data/tips.csv',header=0)

1. Đọc dữ liệu


In [37]:
tip.head(10)

Unnamed: 0  total_bill   tip     sex smoker  day    time  size
0           0       16.99  1.01  Female     No  Sun  Dinner     2
1           1       10.34  1.66    Male     No  Sun  Dinner     3
2           2       21.01  3.50    Male     No  Sun  Dinner     3
3           3       23.68  3.31    Male     No  Sun  Dinner     2
4           4       24.59  3.61  Female     No  Sun  Dinner     4
5           5       25.29  4.71    Male     No  Sun  Dinner     4
6           6        8.77  2.00    Male     No  Sun  Dinner     2
7           7       26.88  3.12    Male     No  Sun  Dinner     4
8           8       15.04  1.96    Male     No  Sun  Dinner     2
9           9       14.78  3.23    Male     No  Sun  Dinner     2

2. Xoá cột `Unnamed: 0`


In [38]:
tip.drop(columns=['Unnamed: 0'],inplace=True)

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [40]:
tip.shape

(244, 7)

4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [41]:
tip.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [67]:
Female=tip[tip['sex']=='Female']
Male=tip[tip['sex']=='Male']
print(f"số lượng nu là: {len(Female)}, Tỷ lệ: {np.round(len(Female)/tip['sex'].count(),2)}")
print(f"số lượng nu là: {len(Male)}, Tỷ lệ: {np.round(len(Male)/tip['sex'].count(),2)}")



số lượng nu là: 87, Tỷ lệ: 0.36
số lượng nu là: 157, Tỷ lệ: 0.64


6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [68]:
yesSmoler=tip[tip['smoker']=='Yes']
print(f"Số lượng người hút thuốc là: {len(yesSmoler)},Tỷ lệ: {np.round(len(yesSmoler)/tip['smoker'].count(),2)}")

Số lượng người hút thuốc là: 93,Tỷ lệ: 0.38


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [72]:
day=tip['day'].value_counts()
print(f"Ngày nào trong tuần khách hàng hay đến dùng bữa nhất là: {day.index[0]}, Tỷ lệ: {np.round(day[0]/tip['day'].count(),2)}")

Ngày nào trong tuần khách hàng hay đến dùng bữa nhất là: Sat, Tỷ lệ: 0.36


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [75]:
time=tip['time'].value_counts()
print(f"Thời gian khách hàng hay đến dùng bữa nhất là: {time.index[0]}")

Thời gian khách hàng hay đến dùng bữa nhất là: Dinner


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [81]:
size=tip['size'].value_counts()
print(f"Nhóm khách dùng bữa nhiều nhất là: {size.index[0]},số lượng là:{len(size)}")

Nhóm khách dùng bữa nhiều nhất là: 2,số lượng là:6


10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [87]:
print(f"Trung bình trong một bữa ăn nhà hàng phục vụ {np.round(tip['size'].mean(),0)} khách")

Trung bình trong một bữa ăn nhà hàng phục vụ 3.0 khách


---